Libplanet/Store/LiteDBStore.cs

```diff
@@ -0,0 +1,631 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security.Cryptography;
+using Libplanet.Action;
+using Libplanet.Blocks;
+using Libplanet.Serialization;
+using Libplanet.Tx;
+using LiteDB;
+
+namespace Libplanet.Store
+{
+    public class LiteDBStore : IStore, IDisposable
+    {
+        private readonly LiteDatabase _db;
+
+        public LiteDBStore(string path)
+        {
+            if (path is null)
+            {
+                throw new ArgumentNullException(nameof(path));
+            }
+
+            var connectionString = new ConnectionString
+            {
+                Filename = path,
+                Journal = false,
+                CacheSize = 50000,
+            };
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) &&
+                Type.GetType("Mono.Runtime") is null)
+            {
+                // macOS + .NETCore doesn't support shared lock.
+                connectionString.Mode = LiteDB.FileMode.Exclusive;
+            }
+
+            _db = new LiteDatabase(connectionString);
+            _db.Mapper.RegisterType(
+                hash => hash.ToByteArray(),
+                b => new HashDigest<SHA256>(b));
+            _db.Mapper.RegisterType(
+                txid => txid.ToByteArray(),
+                b => new TxId(b));
+            _db.Mapper.RegisterType(
+                address => address.ToByteArray(),
+                b => new Address(b.AsBinary));
+        }
+
+        private LiteCollection<TxIdDoc> StagedTxIds =>
+            _db.GetCollection<TxIdDoc>("staged_txids");
+
+        /// <inheritdoc/>
+        public IEnumerable<string> ListNamespaces()
+        {
+            return _db.GetCollectionNames()
+                .Where(name => name.StartsWith("indices_"))
+                .Select(name => name.Split('_')[1]);
```

Comment:
How about this?

```csharp
const string prefix = "indices_";
readonly int prefixLength = prefix.Length;
return _db.GetCollectionNames()
    .Where(name => name.StartsWith(prefix))
    .Select(name => name.Substring(prefixLength));
```

Reply:
I'll refactor with some class level constant. (e.g. `const string _indexPrefix`).